### ISSUE TITLE
Uncrustify sometimes removes return statements

### ISSUE THREAD

**Issue by tidalscale-keithr** [uncrustify/uncrustify#3757]
[//]: # "  If the issue is connected to existing Uncrustify options please, if possible, add the"
[//]: # "  following information to ease up the process:"
[//]: # "  • a link to a debug file:"
[//]: # "      generated with: 'uncrustify -p debug.txt -c pathToUsedConfig.cfg -f toBeFormatedFile.cpp' "
[//]: # "      Example: [debug.txt](https://linkToTheFile)"
[//]: # "      Example hosters for debug files: pastebin.com, gist.github.com, ..."
[//]: # "      The used config file is included in the debug file and does not need to be included here."
[//]: # 
[//]: # "  • include a small but complete test file that will be uncrustifyed"
[//]: # "  • include the generated results"
[//]: # "  • include the expected results"
[//]: # 
[//]: # "  ✋ please add a line containing ``` above and below of each of those three code sections"
[//]: # 
[//]: # "  • include the current version of your Uncrustify executable"
[//]: # "      printout via 'uncrustify -v'"
[//]: # "      Example: current version: uncrustify 0.63"
[//]: # "      or if possible additionally with the git sha of the commit"
[//]: # "                    current version: uncrustify 0.63 dc7b412"
[//]: # 
[//]: # "  • if possible include a version that worked"
[//]: # "      Example: working version: uncrustify 0.63"
[//]: # "      or"
[//]: # "                    working version: uncrustify 0.63 2a5e88f"
Starting with Uncrustify-0.71.0 and continuing to the current git revision (Uncrustify-0.75.0-200-97d51e7a), uncrustify removes bare return statements that are the single-line body of an if statement. Uncrustify-0.70.1 does not have the problem.

```
void
function(int skip_it)
{
    if (skip_it)
        return;

    do_it();
}
```

Becomes

```
void
function(int skip_it)
{
    if (skip_it)


    do_it();
}
```

In this case the expected results should look the same as the input.

[debug.txt](https://gist.github.com/tidalscale-keithr/e019e02d9ea4049a38a7b33a6dfa31dd.js)

**Comment by guy-maurel** [comment]
debug.txt is NOT a text.
Please, join a text-only file

**Comment by guy-maurel** [comment]
The current git version works pretty fine under Linux Mint.
  https://github.com/uncrustify/uncrustify
Please test it and report!
```.txt
Uncrustify_d-0.75.0-210-05d722c7
do_source_file(1509): Parsing: tidalscale-keithr.cpp as language CPP
void
function(skip_it)
{
        if (skip_it)
                return;

        do_it();
}

```

**Comment by tidalscale-keithr** [comment]
Apparently gist.github.com messed up the debug.txt. Here it is just attached to this comment.

I have tested with the current git version, but it isn't Uncrustify_d-0.75.0-210-05d722c7. When I pull the git repo I don't see a `05d722c7` commit at all. The tip of git I see is Uncrustify-0.75.0-205-dab75a2b, and it still has the problem. I've tested on freebsd 13, void linux, and Mac OS Big Sur; all of them show the problem.

[debug.txt](https://github.com/uncrustify/uncrustify/files/9169914/debug.txt)

**Comment by micheleCTDE** [comment]
I can also confirm that current uncrustify version with default config works fine with the example above.

@tidalscale-keithr can you attach your config?

**Comment by tidalscale-keithr** [comment]
I've just done another `git pull` and tested with the latest commit, `ae1ceafb`. The problem still occurs.

My config is attached. The latest version complains about a deprecated option, but since this same config has caused this problem for several versions, most of which don't complain about the config, that can't be the issue.

[tidalscale.cfg.txt](https://github.com/uncrustify/uncrustify/files/9215819/tidalscale.cfg.txt)

**Comment by alexhenrie** [comment]
I can confirm the bug: It happens when `mod_remove_empty_return=true`. The first bad commit is 04f28db0d606ec8257a428f2fcc2f3af97229ade, "don't remove the return statement".

**Comment by micheleCTDE** [comment]
>My config is attached. The latest version complains about a deprecated option, but since this same config has caused this problem for several versions, most of which don't complain about the config, that can't be the issue.

@tidalscale-keithr thanks, I can also confirm the issue with your config

**Comment by micheleCTDE** [comment]
This has been fixed by PR #3958.
@tidalscale-keithr could you test and confirm we can close the issue?
